esmvaltool/diag_scripts/climate_metrics/tcre.py

```diff
@@ -286,8 +286,10 @@ def _get_plot_kwargs(all_plot_kwargs: dict, group: str) -> dict:
 def _load_and_preprocess_data(cfg: dict) -> list[dict]:
     """Load and preprocess data."""
     input_data = list(cfg["input_data"].values())
+    changed_input_data = []
 
     for dataset in input_data:
+        dataset = dict(dataset)
```

Comment:
I think this is the only line we need

Reply:
sadly no, just a dict copy was the first thing I tried yesterday, still tossing a SegFault

Reply:
that'll do! Let me put that in with a comment too

Reply:
done in b4fc0ec1d7604ab52b362f7049aaeb2f3dd356b1